pkg/provider/cloudru/secretmanager/client.go

```diff
@@ -0,0 +1,182 @@
+	AccessSecretVersionByPath(ctx context.Context, projectID, path string, version *int32) ([]byte, error)
+	// AccessSecretVersion gets the secret by the given request.
+	AccessSecretVersion(ctx context.Context, id, version string) ([]byte, error)
+}
+
+// Client is a client for the Cloud.ru Secret Manager.
+type Client struct {
+	apiClient SecretProvider
+
+	projectID string
+}
+
+// GetSecret gets the secret by the remote reference.
+func (c *Client) GetSecret(ctx context.Context, ref esv1beta1.ExternalSecretDataRemoteRef) ([]byte, error) {
+	secret, err := c.accessSecret(ctx, ref.Key, ref.Version)
+	if err != nil {
+		return nil, err
+	}
+
+	prop := strings.TrimSpace(ref.Property)
+	if prop == "" {
+		return secret, nil
+	}
+
+	// For more obvious behavior, we return an error if we are dealing with invalid JSON
+	// this is needed, because the gjson library works fine with value for `key`, for example:
+	//
+	// {"key": "value", another: "value"}
+	//
+	// but it will return "" when accessing to a property `another` (no quotes)
+	if err = json.Unmarshal(secret, &map[string]interface{}{}); err != nil {
+		return nil, fmt.Errorf("expecting the secret %q in JSON format, could not access property %q", ref.Key, ref.Property)
+	}
+
+	result := gjson.Parse(string(secret)).Get(prop)
+	if !result.Exists() {
+		return nil, fmt.Errorf("the requested property %q does not exist in secret %q", prop, ref.Key)
+	}
+
+	return []byte(result.Str), nil
+}
+
+func (c *Client) GetSecretMap(ctx context.Context, ref esv1beta1.ExternalSecretDataRemoteRef) (map[string][]byte, error) {
+	secret, err := c.accessSecret(ctx, ref.Key, ref.Version)
+	if err != nil {
+		return nil, err
+	}
+
+	secretMap := make(map[string]json.RawMessage)
+	if err = json.Unmarshal(secret, &secretMap); err != nil {
+		return nil, fmt.Errorf("expecting the secret %q in JSON format", ref.Key)
+	}
+
+	out := make(map[string][]byte)
+	for k, v := range secretMap {
+		out[k] = []byte(strings.Trim(string(v), "\""))
+	}
+
+	return out, nil
+}
+
+// GetAllSecrets gets all secrets by the remote reference.
+func (c *Client) GetAllSecrets(ctx context.Context, ref esv1beta1.ExternalSecretFind) (map[string][]byte, error) {
+	if len(ref.Tags) == 0 && ref.Name == nil {
+		return nil, fmt.Errorf("at least one of the following fields must be set: tags, name")
+	}
+
+	var nameFilter string
+	if ref.Name != nil {
+		nameFilter = ref.Name.RegExp
+	}
+
+	searchReq := &adapter.ListSecretsRequest{
+		ProjectID: c.projectID,
+		Labels:    ref.Tags,
+		NameRegex: nameFilter,
+	}
+	secrets, err := c.apiClient.ListSecrets(ctx, searchReq)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list secrets: %w", err)
+	}
+
+	out := make(map[string][]byte)
+	for _, s := range secrets {
+		secret, err := c.GetSecretMap(ctx, esv1beta1.ExternalSecretDataRemoteRef{Key: s.GetId()})
+		if err != nil {
+			return nil, err
+		}
+
+		for k, v := range secret {
+			out[k] = v
+		}
+	}
```

Comment:
The expected behavior of `GetAllSecrets` method is to return key,value pairs containing the full secret. So, if `secret1` is composed of `{"foo":"bar"}`, we should return `secret1: {"foo":"bar"}` in this method.